networkx/algorithms/tests/test_euler.py

```diff
@@ -273,3 +273,22 @@ def test_on_complete_graph(self):
         G = nx.complete_graph(4)
         assert nx.is_eulerian(nx.eulerize(G))
         assert nx.is_eulerian(nx.eulerize(nx.MultiGraph(G)))
+
+    def test_on_non_eulerian_graph(self):
+        G = nx.cycle_graph(18)
+        G.add_edge(0, 18)
+        G.add_edge(18, 19)
+        G.add_edge(17, 19)
+        G.add_edge(4, 20)
+        G.add_edge(20, 21)
+        G.add_edge(21, 22)
+        G.add_edge(22, 23)
+        G.add_edge(23, 24)
+        G.add_edge(24, 25)
+        G.add_edge(25, 26)
+        G.add_edge(26, 27)
+        G.add_edge(27, 28)
+        G.add_edge(28, 13)
+        G = nx.eulerize(G)
+        assert nx.is_eulerian(nx.eulerize(G))
+        assert nx.number_of_edges(G) == 39
```

Comment:
Maybe you should check that G is not eulerian first.

And you might consider making `EG = nx.eulerize(G)` so you can refer to each later.
You want to test that G is not eulerian and that EG is eulerian.

As written, it is not clear to me why in line 293 you need `nx.eulerize(G)` inside the `nx.is_eulerian`. Hasn't G already been eulerized?

Reply:
> As written, it is not clear to me why in line 293 you need nx.eulerize(G) inside the nx.is_eulerian. Hasn't G already been eulerized?

Yes. Fixed.